ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -358,5 +375,32 @@ public override void OnReactInstanceDispose()
 
             return null;
         }
+
+        private async Task<bool> IsCancelledOrDisposed()
+        {
+            try
+            {
+                var token = _cancelSource.Token;
+                await _mutex.WaitAsync(token).ConfigureAwait(false);
+            }
+            catch (OperationCanceledException)
+            {
```

Comment:
can we add a `when` clause to this so we only catch when the OperationCanceledException is sourced from the correct cancellation token?